app/controllers/vendor_api/confirm_candidate_enrolment_controller.rb

```diff
@@ -9,12 +9,14 @@ def confirm
       raise InvalidMetadata unless Metadata.new(params[:meta]).valid?
 
       application_choice = ApplicationChoice.find(params[:application_id])
-      result = ConfirmEnrolment.new(application_choice: application_choice).call
+      confirmation = ConfirmEnrolment.new(application_choice: application_choice)
 
-      if result.successful?
+      if confirmation.save
         render json: {
-          data: SingleApplicationPresenter.new(result.application_choice).as_json,
+          data: SingleApplicationPresenter.new(application_choice).as_json,
```

Comment:
Should we `.reload` here (and in the other controllers)? the state that was loaded at line 11 will hopefully not be the same as the state after update

Reply:
I'm assuming the object is updated together with the database. As long as the test still passes I don't tink we have to do anything more:

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/b3e547734a18028b7b98d67b2b2fa91505a3f161/spec/requests/vendor_api/confirm_enrolment_spec.rb#L18-L27